treeherder/model/management/commands/cycle_data.py

```diff
@@ -89,16 +103,24 @@ def prune(id_name, model):
 
 
 class PerfherderCycler(DataCycler):
+    DEFAULT_MAX_RUNTIME = timedelta(hours=23)
+
+    @has_valid_explicit_days
     def __init__(
         self,
         chunk_size: int,
         sleep_time: int,
         is_debug: bool = None,
+        days: int = None,
+        max_runtime: timedelta = None,
         strategies: List[RemovalStrategy] = None,
         **kwargs,
     ):
         super().__init__(chunk_size, sleep_time, is_debug)
-        self.strategies = strategies or RemovalStrategy.fabricate_all_strategies(chunk_size)
+        self.max_runtime = max_runtime or PerfherderCycler.DEFAULT_MAX_RUNTIME
```

Comment:
Here you can use the new class `MaxRuntime`

Reply:
Good catch! 🎣